### PR TITLE
Version value rework and centralization

### LINKED ISSUE #92: Version increments
We have two versions that exist in the application:
[setup.py](https://github.com/eth-educators/ethstaker-deposit-cli/blob/main/setup.py) and [settings.py](https://github.com/eth-educators/ethstaker-deposit-cli/blob/main/ethstaker_deposit/settings.py#L4)

We can manually update these each time we release or we can use version tools to help automate it.

### PR DESCRIPTION
**Pull request by remyroy**
Fixes #92

## Changes

- Centralize version in the `__version__` variable in the `__init__.py` file.
- Rework version in pyproject.toml to use the dynamic value in `__version__`.
- Use the `__version__` in `settings.py` file.
- Adds a `--version` option to output the tool version and exit.

#### What types of changes does your code introduce?

- [X] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [X] Refactoring
- [ ] Documentation update
- [X] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

#### Notes on testing

Builds should be tested and tests suite run again.

#### Requires documentation update

- [ ] Yes
- [X] No

#### Requires explanation in Release Notes

- [ ] Yes
- [X] No

## Remarks

The idea is that `__version__` will contain the single source of truth for the version of the project. As we perform a release, that file will be updated to include the new version. As we start the development for the next version, it should be updated again with a new version number that ends with `.dev`.

### CODE REVIEW

**Comment by yorickdowne** [comment]
`Couldn't combine from non-existent path 'coverage*'` - did this break coverage reporting?

**Comment by remyroy** [comment]
Let me see what's wrong with the coverage CI stuff.

**Comment by remyroy** [comment]
The coverage issue seems unrelated to this PR. Here is an example of another run where the coverage report doesn't work: https://github.com/eth-educators/ethstaker-deposit-cli/actions/runs/10659330561

**Comment by remyroy** [comment]
Coverage issue has been descrived in #137

**Comment by yorickdowne** [comment]
Do we still have the version in pyproject.toml as well?

File: ethstaker_deposit/deposit.py [hunk 1]
```diff
@@ -100,6 +103,7 @@ def list_commands(self, ctx: click.Context) -> list[str]:
     ),
     hidden=False,
 )
+@click.version_option(version=DEPOSIT_CLI_VERSION)
```
Comment (valefar-on-discord) [inline ethstaker_deposit/deposit.py:106]: Nice touch 👍

**Review by valefar-on-discord — APPROVED** [review]

**Pull request merged**